docs/amended-alias-resolution.md

```diff
@@ -0,0 +1,165 @@
+# Amended Alias Syntax and Resolution Semantics
+
+## Summary
+
+We need to finalize a syntax for aliases in require statements and determine intuitive resolution semantics as we prepare to build a package management system.
+
+## Motivation
+
+As we prepare to build an official Luau package management system, we need unambiguous alias functionality.
+
+In an [old RFC for require-by-string](https://github.com/luau-lang/luau/pull/969), we removed the `@` prefix from aliases altogether and opted for the following design:
+the path `require("my/module")` is resolved by first checking if `my` is an alias defined in a `.luaurc` file.
+If so, we perform a string substitution on `my`.
+If not, we try to resolve `my/module` relative to the requiring file.
+
+This design changed in the next few months and became what we now have in [require-by-string-aliases.md](require-by-string-aliases.md):
+when requiring an alias, it must be prepended with the `@` prefix.
+Otherwise, we will assume the given string does not contain an alias and will resolve it relative to the requiring file.
+
+There have been disagreements about which approach is more appropriate, so this RFC will serve as a way to document the ongoing discussion and the final decision we make.
+
+## Design and Drawbacks
+
+Below, we have multiple different options for syntax and resolution semantics.
+For each design, assume that the following `.luaurc` file is defined:
+```json
+{
+    "aliases": {
+        "libs": "/My/Libraries/Directory"
+    }
+}
+```
+Additionally, assume that we have a file named `dependency.luau` located in these two locations:
+1. `./libs/dependency.luau`
+2. `/My/Libraries/Directory/dependency.luau`
+
+### (1) Make explicit prefixes necessary
+
+#### (1a) Make aliases explicit with `@` prefix
+
+We make usage of the `@` prefix necessary in aliased require statements.
+This means that any unprefixed string is always treated as an unaliased path.
+```luau
+-- Requires ./libs/dependency.luau
+require("libs/dependency")
+
+-- Requires ./libs/dependency.luau
+require("./libs/dependency")
+
+-- Requires /My/Libraries/Directory/dependency.luau
+require("@libs/dependency")
+```
+
+The main drawback of this approach is that projects with many _external_ dependencies will have many `@` symbols appearing within require statements.
+
+This is similar to [Lune's approach](https://lune-org.github.io/docs/getting-started/2-introduction/8-modules#file-require-statements) and is identical to the approach currently described in [require-by-string-aliases.md](require-by-string-aliases.md).
+
+#### (1b) Make relative paths explicit with either `./` or `../` prefix
+
+We make usage of either the `./` or `../` prefix necessary in relative require statements.
```

Comment:
I personally like this because it resolves the footgun with missing `@` and for a while we actually had this. I hope we remove support for require by implicit relative path.

Reply:
I believe both (1b) and (1c) force relative paths to be explicitly specified. I thought for a moment that (3a) also supported this, but I was missing a use case that I just added. So, (3a) also supports implicit relative paths.